README.md

```diff
@@ -1,4 +1,1060 @@
+
+### 3.1.4. `nixConfig`
+
+Unless you know what you are doing, you should not change `nixConfig`.
+
+You could always add new `extra-substituters` and `extra-trusted-public-keys`, but do not delete the existing ones, or you won't have access to IOG caches. 
+
+For the caches to work properly, it is sufficient that the following two lines are included in your `/etc/nix/nix.conf`:
+```txt
+trusted-users = USER
+experimental-features = nix-command flakes
+```
+Replace `USER` with the result of running `whoami`. 
+
+You may need to reload the nix daemon on Darwin for changes to `/etc/nix/nix.conf` to take effect:
+```bash
+sudo launchctl stop org.nixos.nix-daemon
+sudo launchctl start org.nixos.nix-daemon
+```
+Leave `allow-import-from-derivation` set to `true` for `haskell.nix` for work correctly.
+
+## 3.2. `nix/iogx-config.nix`
+
+```nix
+# Example of a valid nix/iogx-config.nix
+{ 
+  repoRoot = ../.; 
+  systems = [ "x86_64-linux" "x86_64-darwin" ]; 
+  haskellCompilers = [ "ghc8107" ]; 
+  defaultHaskellCompiler = "ghc8107"; 
+  shouldCrossCompile = true; 
+} 
+```
+
+This file must exist and it contains the main configuration values for your project.
+
+### 3.2.1. `repoRoot` 
+
+A Nix path to an existing folder containing your `cabal.project` file.
+
+In future versions this field will be removed and it will default to the repository top-level directory, but until then it must be explicitly set (usually to `../.`).
+
+### 3.2.2. `systems`
+
+The non-empty list of systems against which your project can be built. 
+
+These are standard Nix values found in `nixpkgs.stdenv.system`.
+
+This field affects your final [flake outputs](#310-flake-outputs-format).
+
+This field is required.
+
+### 3.2.3. `haskellCompilers`
+
+The non-empty list of GHC versions that can build your project. 
+
+Currently two GHC versions are supported and provided by IOGX: `ghc8107` and `ghc927`.
+
+This field affects your final [flake outputs](#310-flake-outputs-format).
+
+This field is required.
+
+### 3.2.4. `defaultHaskellCompiler`
+
+Only one compiler at a time can be visible in the `$PATH` in the shell.
+
+When calling `nix develop`, the `defaultHaskellCompiler` will be selected. 
+
+To enter a shell with a different compiler, see to the [flake outputs](#310-flake-outputs-format) section. 
+
+This field is optional and defaults to the first (leftmost)  compiler in [`haskellCompilers`](#323-haskellcompilers).
+
+### 3.2.5. `shouldCrossCompile`
+
+Cross-compilation on Windows is available via `mingwW64` on `x86_64-linux` only. 
+
+If your project cannot be cross-compiled then set this field to `false`. 
+
+This field affects your final [flake outputs](#310-flake-outputs-format).
+
+This field is optional and defaults to `true`.
+
+## 3.3. `nix/haskell-project.nix`
+
+```nix
+# Default nix/haskell-project.nix
+{ inputs 
+, inputs' 
+, pkgs 
+, { haskellCompiler
+  , enableHaddock 
+  , enableProfiling 
+  , enableCross 
+  }@meta 
+}:
```

Comment:
If I don't use the args can I omit the function?

Reply:
Good idea -> changed for all files